soc/arm/st_stm32/stm32f7/soc.c

```diff
@@ -32,11 +32,6 @@ static int st_stm32f7_init(const struct device *arg)
 
 	key = irq_lock();
 
-	SCB_EnableICache();
```

Comment:
This is not correct at all. The code in `arch/arm/core/aarch32/cortex_m/scb.c` is run very early in the boot process, way before the STM32F7 specific code is used. Its goal is to bring the hardware in a know, reproducible state.

There is still need to enable it later during the boot process in the STM32F7 specific code.

Reply:
ok, restored cache enable if needed in the stm32f7/soc.c 
add the config for testing the stm32f746zg nucleo board only

Reply:
First of all the I-Cache can be kept enabled in that case, it doesn't interfere with DMA.

Secondly, I still do not get why we want to disable the cache if CONFIG_NOCACHE_MEMORY is not enabled. It has a huge performance impact, I would see that the other way around, i.e. if that CONFIG_UART_ASYNC_API selects CONFIG_NOCACHE_MEMORY on Cortex M7 or STM32F7.

Alternatively we might want to always enable CONFIG_NOCACHE_MEMORY on Cortex M7 or STM32F7, or enable it when DMA is enabled.